fix: drop support for projects classic

mikey- opened the pull request:
<!--

  👋 Hi there!
  Thanks for contributing to metrics and helping us to improve!

  Please:
    - Read CONTRIBUTING.md first 
    - Check you're not duplicating another existing pull request
    - Provide a clear and concise description

  Note that:
    - Your code will be automatically formatted by github-actions
    - Head branches are automatically deleted when merged

-->

Fix: https://github.com/lowlighter/metrics/issues/1706 by dropping support for projects classic. 

As mentioned in the issue description:
> The "Manager" achievement (from plugin_achievements) is failing because it uses the deprecated Projects (classic) API: [sunset on May 23, 2024](https://github.blog/changelog/2024-05-23-sunset-notice-projects-classic/).

These changes fix this error by removing legacy API calls/GraphQL queries and updating those API calls/GraphQL queries to use Projects V2. Most of the work was done by @stalomeow, with a few additions from myself. It is my understanding that this is unlikely to be merged, but it's worth a shot, since this is such a great and fun repo.

mikey- committed:
fix: drop projects classic support

mikey- committed:
fix: drop projects classic support

mikey- committed:
self rely this ish

dependabot[bot] committed:
chore(deps): bump the npm_and_yarn group with 31 updates

Bumps the npm_and_yarn group with 32 updates:

| Package | From | To |
| --- | --- | --- |
| [axios](https://github.com/axios/axios) | `1.5.0` | `1.9.0` |
| [@googlemaps/google-maps-services-js](https://github.com/googlemaps/google-maps-services-js) | `3.3.38` | `3.4.1` |
| [ejs](https://github.com/mde/ejs) | `3.1.9` | `3.1.10` |
| [express](https://github.com/expressjs/express) | `4.18.2` | `4.21.2` |
| [prismjs](https://github.com/PrismJS/prism) | `1.29.0` | `1.30.0` |
| [sanitize-html](https://github.com/apostrophecms/sanitize-html) | `2.11.0` | `2.12.1` |
| [sharp](https://github.com/lovell/sharp) | `0.32.3` | `0.32.6` |
| [vue](https://github.com/vuejs/core) | `2.7.14` | `3.0.0` |
| [libxmljs2](https://github.com/marudor/libxmljs2) | `0.32.0` | `0.35.0` |
| [@babel/traverse](https://github.com/babel/babel/tree/HEAD/packages/babel-traverse) | `7.22.17` | `7.27.1` |
| [@octokit/request](https://github.com/octokit/request.js) | `8.1.1` | `8.4.1` |
| [@actions/github](https://github.com/actions/toolkit/tree/HEAD/packages/github) | `5.1.1` | `6.0.0` |
| [@octokit/request-error](https://github.com/octokit/request-error.js) | `2.1.0` | `5.1.1` |
| [@octokit/plugin-paginate-rest](https://github.com/octokit/plugin-paginate-rest.js) | `2.21.3` | `8.0.0` |
| [body-parser](https://github.com/expressjs/body-parser) | `1.20.1` | `1.20.3` |
| [braces](https://github.com/micromatch/braces) | `3.0.2` | `3.0.3` |
| [cookie](https://github.com/jshttp/cookie) | `0.5.0` | `0.7.1` |
| [crypto-js](https://github.com/brix/crypto-js) | `4.1.1` | `4.2.0` |
| [follow-redirects](https://github.com/follow-redirects/follow-redirects) | `1.15.2` | `1.15.9` |
| [ip](https://github.com/indutny/node-ip) | `1.1.8` | `removed` |
| [pac-resolver](https://github.com/TooTallNate/proxy-agents/tree/HEAD/packages/pac-resolver) | `7.0.0` | `7.0.1` |
| [nanoid](https://github.com/ai/nanoid) | `3.3.6` | `3.3.11` |
| [path-to-regexp](https://github.com/pillarjs/path-to-regexp) | `0.1.7` | `0.1.12` |
| [postcss](https://github.com/postcss/postcss) | `8.4.28` | `8.5.3` |
| [tar-fs](https://github.com/mafintosh/tar-fs) | `2.1.1` | `3.0.8` |
| [tar-fs](https://github.com/mafintosh/tar-fs) | `3.0.4` | `3.0.8` |
| [puppeteer](https://github.com/puppeteer/puppeteer) | `21.2.1` | `24.8.0` |
| [send](https://github.com/pillarjs/send) | `0.18.0` | `0.19.0` |
| [serve-static](https://github.com/expressjs/serve-static) | `1.15.0` | `1.16.2` |
| [tar](https://github.com/isaacs/node-tar) | `6.1.15` | `6.2.1` |
| [undici](https://github.com/nodejs/undici) | `5.23.0` | `5.29.0` |
| [ws](https://github.com/websockets/ws) | `8.14.1` | `8.18.2` |

Updates `axios` from 1.5.0 to 1.9.0
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.5.0...v1.9.0)

Updates `@googlemaps/google-maps-services-js` from 3.3.38 to 3.4.1
- [Release notes](https://github.com/googlemaps/google-maps-services-js/releases)
- [Changelog](https://github.com/googlemaps/google-maps-services-js/blob/master/CHANGELOG.md)
- [Commits](https://github.com/googlemaps/google-maps-services-js/compare/v3.3.38...v3.4.1)

Updates `ejs` from 3.1.9 to 3.1.10
- [Release notes](https://github.com/mde/ejs/releases)
- [Commits](https://github.com/mde/ejs/compare/v3.1.9...v3.1.10)

Updates `express` from 4.18.2 to 4.21.2
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/4.21.2/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.21.2)

Updates `prismjs` from 1.29.0 to 1.30.0
- [Release notes](https://github.com/PrismJS/prism/releases)
- [Changelog](https://github.com/PrismJS/prism/blob/v2/CHANGELOG.md)
- [Commits](https://github.com/PrismJS/prism/compare/v1.29.0...v1.30.0)

Updates `sanitize-html` from 2.11.0 to 2.12.1
- [Changelog](https://github.com/apostrophecms/sanitize-html/blob/main/CHANGELOG.md)
- [Commits](https://github.com/apostrophecms/sanitize-html/compare/2.11.0...2.12.1)

Updates `sharp` from 0.32.3 to 0.32.6
- [Release notes](https://github.com/lovell/sharp/releases)
- [Changelog](https://github.com/lovell/sharp/blob/v0.32.6/docs/changelog.md)
- [Commits](https://github.com/lovell/sharp/compare/v0.32.3...v0.32.6)

Updates `vue` from 2.7.14 to 3.0.0
- [Release notes](https://github.com/vuejs/core/releases)
- [Changelog](https://github.com/vuejs/core/blob/v3.0.0/CHANGELOG.md)
- [Commits](https://github.com/vuejs/core/commits/v3.0.0)

Updates `libxmljs2` from 0.32.0 to 0.35.0
- [Release notes](https://github.com/marudor/libxmljs2/releases)
- [Commits](https://github.com/marudor/libxmljs2/compare/v0.32.0...v0.35.0)

Updates `@babel/traverse` from 7.22.17 to 7.27.1
- [Release notes](https://github.com/babel/babel/releases)
- [Changelog](https://github.com/babel/babel/blob/main/CHANGELOG.md)
- [Commits](https://github.com/babel/babel/commits/v7.27.1/packages/babel-traverse)

Updates `@octokit/request` from 8.1.1 to 8.4.1
- [Release notes](https://github.com/octokit/request.js/releases)
- [Commits](https://github.com/octokit/request.js/compare/v8.1.1...v8.4.1)

Updates `@actions/github` from 5.1.1 to 6.0.0
- [Changelog](https://github.com/actions/toolkit/blob/main/packages/github/RELEASES.md)
- [Commits](https://github.com/actions/toolkit/commits/HEAD/packages/github)

Updates `@octokit/request-error` from 2.1.0 to 5.1.1
- [Release notes](https://github.com/octokit/request-error.js/releases)
- [Commits](https://github.com/octokit/request-error.js/compare/v2.1.0...v5.1.1)

Updates `@octokit/plugin-paginate-rest` from 2.21.3 to 8.0.0
- [Release notes](https://github.com/octokit/plugin-paginate-rest.js/releases)
- [Commits](https://github.com/octokit/plugin-paginate-rest.js/compare/v2.21.3...v8.0.0)

Updates `body-parser` from 1.20.1 to 1.20.3
- [Release notes](https://github.com/expressjs/body-parser/releases)
- [Changelog](https://github.com/expressjs/body-parser/blob/master/HISTORY.md)
- [Commits](https://github.com/expressjs/body-parser/compare/1.20.1...1.20.3)

Updates `braces` from 3.0.2 to 3.0.3
- [Changelog](https://github.com/micromatch/braces/blob/master/CHANGELOG.md)
- [Commits](https://github.com/micromatch/braces/compare/3.0.2...3.0.3)

Updates `cookie` from 0.5.0 to 0.7.1
- [Release notes](https://github.com/jshttp/cookie/releases)
- [Commits](https://github.com/jshttp/cookie/compare/v0.5.0...v0.7.1)

Updates `crypto-js` from 4.1.1 to 4.2.0
- [Commits](https://github.com/brix/crypto-js/compare/4.1.1...4.2.0)

Updates `follow-redirects` from 1.15.2 to 1.15.9
- [Release notes](https://github.com/follow-redirects/follow-redirects/releases)
- [Commits](https://github.com/follow-redirects/follow-redirects/compare/v1.15.2...v1.15.9)

Removes `ip`

Updates `pac-resolver` from 7.0.0 to 7.0.1
- [Release notes](https://github.com/TooTallNate/proxy-agents/releases)
- [Changelog](https://github.com/TooTallNate/proxy-agents/blob/main/packages/pac-resolver/CHANGELOG.md)
- [Commits](https://github.com/TooTallNate/proxy-agents/commits/pac-resolver@7.0.1/packages/pac-resolver)

Updates `nanoid` from 3.3.6 to 3.3.11
- [Release notes](https://github.com/ai/nanoid/releases)
- [Changelog](https://github.com/ai/nanoid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/ai/nanoid/compare/3.3.6...3.3.11)

Updates `path-to-regexp` from 0.1.7 to 0.1.12
- [Release notes](https://github.com/pillarjs/path-to-regexp/releases)
- [Changelog](https://github.com/pillarjs/path-to-regexp/blob/master/History.md)
- [Commits](https://github.com/pillarjs/path-to-regexp/compare/v0.1.7...v0.1.12)

Updates `postcss` from 8.4.28 to 8.5.3
- [Release notes](https://github.com/postcss/postcss/releases)
- [Changelog](https://github.com/postcss/postcss/blob/main/CHANGELOG.md)
- [Commits](https://github.com/postcss/postcss/compare/8.4.28...8.5.3)

Updates `tar-fs` from 2.1.1 to 3.0.8
- [Commits](https://github.com/mafintosh/tar-fs/compare/v2.1.1...v3.0.8)

Updates `tar-fs` from 3.0.4 to 3.0.8
- [Commits](https://github.com/mafintosh/tar-fs/compare/v2.1.1...v3.0.8)

Updates `puppeteer` from 21.2.1 to 24.8.0
- [Release notes](https://github.com/puppeteer/puppeteer/releases)
- [Changelog](https://github.com/puppeteer/puppeteer/blob/main/CHANGELOG.md)
- [Commits](https://github.com/puppeteer/puppeteer/compare/puppeteer-v21.2.1...puppeteer-v24.8.0)

Updates `send` from 0.18.0 to 0.19.0
- [Release notes](https://github.com/pillarjs/send/releases)
- [Changelog](https://github.com/pillarjs/send/blob/master/HISTORY.md)
- [Commits](https://github.com/pillarjs/send/compare/0.18.0...0.19.0)

Updates `serve-static` from 1.15.0 to 1.16.2
- [Release notes](https://github.com/expressjs/serve-static/releases)
- [Changelog](https://github.com/expressjs/serve-static/blob/v1.16.2/HISTORY.md)
- [Commits](https://github.com/expressjs/serve-static/compare/v1.15.0...v1.16.2)

Updates `tar` from 6.1.15 to 6.2.1
- [Release notes](https://github.com/isaacs/node-tar/releases)
- [Changelog](https://github.com/isaacs/node-tar/blob/main/CHANGELOG.md)
- [Commits](https://github.com/isaacs/node-tar/compare/v6.1.15...v6.2.1)

Updates `undici` from 5.23.0 to 5.29.0
- [Release notes](https://github.com/nodejs/undici/releases)
- [Commits](https://github.com/nodejs/undici/compare/v5.23.0...v5.29.0)

Updates `ws` from 8.14.1 to 8.18.2
- [Release notes](https://github.com/websockets/ws/releases)
- [Commits](https://github.com/websockets/ws/compare/8.14.1...8.18.2)

---
updated-dependencies:
- dependency-name: axios
  dependency-version: 1.9.0
  dependency-type: direct:production
  dependency-group: npm_and_yarn
- dependency-name: "@googlemaps/google-maps-services-js"
  dependency-version: 3.4.1
  dependency-type: direct:production
  dependency-group: npm_and_yarn
- dependency-name: ejs
  dependency-version: 3.1.10
  dependency-type: direct:production
  dependency-group: npm_and_yarn
- dependency-name: express
  dependency-version: 4.21.2
  dependency-type: direct:production
  dependency-group: npm_and_yarn
- dependency-name: prismjs
  dependency-version: 1.30.0
  dependency-type: direct:production
  dependency-group: npm_and_yarn
- dependency-name: sanitize-html
  dependency-version: 2.12.1
  dependency-type: direct:production
  dependency-group: npm_and_yarn
- dependency-name: sharp
  dependency-version: 0.32.6
  dependency-type: direct:production
  dependency-group: npm_and_yarn
- dependency-name: vue
  dependency-version: 3.0.0
  dependency-type: direct:production
  dependency-group: npm_and_yarn
- dependency-name: libxmljs2
  dependency-version: 0.35.0
  dependency-type: direct:production
  dependency-group: npm_and_yarn
- dependency-name: "@babel/traverse"
  dependency-version: 7.27.1
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: "@octokit/request"
  dependency-version: 8.4.1
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: "@actions/github"
  dependency-version: 6.0.0
  dependency-type: direct:production
  dependency-group: npm_and_yarn
- dependency-name: "@octokit/request-error"
  dependency-version: 5.1.1
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: "@octokit/plugin-paginate-rest"
  dependency-version: 8.0.0
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: body-parser
  dependency-version: 1.20.3
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: braces
  dependency-version: 3.0.3
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: cookie
  dependency-version: 0.7.1
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: crypto-js
  dependency-version: 4.2.0
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: follow-redirects
  dependency-version: 1.15.9
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: ip
  dependency-version:
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: pac-resolver
  dependency-version: 7.0.1
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: nanoid
  dependency-version: 3.3.11
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: path-to-regexp
  dependency-version: 0.1.12
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: postcss
  dependency-version: 8.5.3
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: tar-fs
  dependency-version: 3.0.8
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: tar-fs
  dependency-version: 3.0.8
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: puppeteer
  dependency-version: 24.8.0
  dependency-type: direct:production
  dependency-group: npm_and_yarn
- dependency-name: send
  dependency-version: 0.19.0
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: serve-static
  dependency-version: 1.16.2
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: tar
  dependency-version: 6.2.1
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: undici
  dependency-version: 5.29.0
  dependency-type: indirect
  dependency-group: npm_and_yarn
- dependency-name: ws
  dependency-version: 8.18.2
  dependency-type: indirect
  dependency-group: npm_and_yarn
...

Signed-off-by: dependabot[bot] <support@github.com>

Alexdelia commented on Dockerfile:
```diff
@@ -10,7 +10,7 @@ RUN chmod +x /metrics/source/app/action/index.mjs \
   # Based on https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md#running-puppeteer-in-docker
   && apt-get update \
-  && apt-get install -y wget gnupg ca-certificates libgconf-2-4 \
+  && apt-get install -y xz-utils wget gnupg ca-certificates libgconf-2-4 \
```
why adding `xz-utils`?

stalomeow replied:
see #1677

mikey- replied:
Great question @Alexdelia: I tried to get away without it, only for the build to fail here: https://github.com/mikey-/metrics/actions/runs/14841494754/job/41665609112#step:4:2057
> /var/lib/gems/3.1.0/gems/mini_portile2-2.8.8/lib/mini_portile2/mini_portile.rb:557:in `xzcat_exe': xzcat not found (RuntimeError)

Then I realised why @stalomeow had made this change in their fork :)

Alexdelia replied:
I see, thanks for the explanation

mikey- replied:
np! Thank YOU for reviewing :) I'll resolve this thread now

mikey- commented:
@lowlighter is there anything stopping this PR from being merged?

felipecrs committed:
fix: notable not showing users for user-owned repositories

felipecrs committed:
Fix CI pipeline failure in forked repository (#1719)

Author: 尤雨东 <root@ltpp.vip>

felipecrs committed:
fix: docker image not building

felipecrs committed:
fix: chrome not found in docker image

felipecrs committed:
Add deno and licensed back

felipecrs committed:
Fix workflow tests with docker image

felipecrs committed:
Avoid reinstalling all dependencies in docker test

mikey- committed:
Merge remote-tracking branch 'felipecrs/master'

mikey- committed:
update actions/github-script to v7

mikey- committed:
update actions/checkout to v4

mikey- committed:
update actions/setup-node to v4

mikey- committed:
updated actions/labeler to v5 & added ci labels

mikey- committed:
updated check-spelling/check-spelling to v0.0.25

mikey- committed:
updated actions/stale to v9

mikey- committed:
updated dessant/lock-threads to v5

mikey- committed:
missed 2 instances of actions/checkout@v3

mikey- committed:
missed 1 instances of actions/setup-node@v3

mikey- committed:
undo changes to .github/scripts/files/examples.yml

mikey- committed:
add myself to "Repository level file" changers list

mikey- committed:
undo changes to .github/workflows/examples.yml

mikey- committed:
see https://github.com/lowlighter/metrics/pull/1715#pullrequestreview-2816097223

mikey- committed:
inclusivity: default branch is 'main'

mikey- committed:
Merge branch 'fix-ci-outdated-actions'

mikey- committed:
updated cpell dictionaries

mikey- committed:
added many extra dictionaries

mikey- committed:
the allow list grows

mikey- committed:
[check-spelling] Update metadata

Update for https://github.com/mikey-/metrics/actions/runs/15459707849/attempts/1
Accepted in https://github.com/mikey-/metrics/pull/3#issuecomment-2942894076

Signed-off-by: check-spelling-bot <check-spelling-bot@users.noreply.github.com>
on-behalf-of: @check-spelling <check-spelling-bot@check-spelling.dev>

mikey- committed:
removed excessive extra dictionaries

mikey- committed:
sync-labels is boolean

mikey- committed:
Merge branch 'fix-ci-outdated-actions'

mikey- committed:
Update check-spelling metadata

mikey- committed:
Merge pull request #1 from mikey-/dependabot-npm_and_yarn-npm_and_yarn-9843fa68b7

chore(deps): bump the npm_and_yarn group with 31 updates

mikey- committed:
trying a different profile on 16personalities test

mikey- committed:
fix typo - replace / for : in cpp extra dictionary

mikey- committed:
Update check-spelling metadata

mikey- committed:
check_extra_dictionaries doesn't work like I thought it did

mikey- committed:
Update check-spelling metadata

mikey- committed:
Update check-spelling metadata

mikey- committed:
update github/codeql-action/init and github/codeql-action/analyze to v3

mikey- commented:
Bump! Please don't close :)

mikey- commented:
The build for this is using deprecated action versions, which is now the reason the for `Check Spelling / Check Spelling (push)` job failures. To address this, I've rebased onto mikey-:fix-ci-outdated-actions. See https://github.com/lowlighter/metrics/pull/1731 for details on changes included.

mikey- commented:
Actually I think the stale label is appropriate. I've tried keeping the branch up to date but there are just too many changes now - kind of lost sight of the incremental change. I will now close this PR and just focus on trying to keep my fork up-to-date and working
